chainermn/communicators/pure_nccl_communicator.py

```diff
@@ -106,58 +111,41 @@ def _allreduce_grad_async(self, model, stream):
             self.gpu_buffer_a.array(n_elems,
                                     dtype=allreduce_grad_dtype),
             stream=stream)
-        self._unpack_params_from_buffer(params, grad_dtype,
+
+        # unpack params from buffer A -> params
+        self._unpack_params_from_buffer(params,
                                         allreduce_grad_dtype, n_elems, stream)
 
-    def _assign_for_allreduce_grad(self, grad_dtype, allreduce_grad_dtype,
-                                   n_elems):
+    def _prepare_allreduce_pack_buffer(self, grad_dtype, allreduce_grad_dtype,
```

Comment:
How about delete `grad_dtype` because it is not used.

Reply:
Thanks!